core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatEntry.scala

```diff
@@ -399,7 +398,7 @@ final class ColumnFormatValue
   override def toData(out: DataOutput): Unit = {
     val buffer = getBufferRetain
     try {
-      val numBytes = buffer.remaining()
+      val numBytes = buffer.limit()
```

Comment:
Ok. We are creating a duplicate buffer during scans. So it should be ok i guess.